### PR TITLE
Changed Bluetooth device discovery on Linux to use Limited Dedicated …

### PR DESCRIPTION
**Pull request by bryangrim**
…Inquiry Access Code (LIAC), since the Third-Party Wiimotes (such as Rock Candy Wiimotes) are not discovered without it.
Also added an accessor function in IONix class to help with checking if the discovered wimmote has already been found.

I have a 'real' wiimote plus two different third-party wiimotes, and with this change all of them get recognized.  (It does not save them between runs of dolphin-emu, which is another thing I would like to investigate.)

(Note that my original pull request is #3230, but due to some problems with commits, pulls, etc. I thought it would be safer to create a new pull request.)

---
This change is [Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3327)

### CODE REVIEW

File: Source/Core/Core/HW/WiimoteReal/IONix.cpp [hunk 1]
```diff
@@ -113,14 +118,16 @@ void WiimoteScanner::FindWiimotes(std::vector<Wiimote*> & found_wiimotes, Wiimot
...
-			//for (int j = 0; j < MAX_WIIMOTES && new_wiimote; ++j)
-			//{
-			//	if (wm[j] && bacmp(&scan_infos[i].bdaddr,&wm[j]->bdaddr) == 0)
-			//		new_wiimote = false;
-			//}
+			for (int j = 0; j < MAX_BBMOTES && new_wiimote; ++j)
+			{
+				// compare this address with the stored addresses in our global array
+				// static_cast is OK here, since we're only ever going to have this subclass in g_wiimotes
+				// on Linux (and likewise, only WiimoteWindows on Windows, etc)
```
Comment (degasus) [inline Source/Core/Core/HW/WiimoteReal/IONix.cpp]: There is a trailing whitespace in this line.

File: Source/Core/Core/HW/WiimoteReal/IONix.cpp [hunk 2]
```diff
@@ -84,9 +83,15 @@ void WiimoteScanner::FindWiimotes(std::vector<Wiimote*> & found_wiimotes, Wiimot
 	inquiry_info scan_infos[max_infos] = {};
 	auto* scan_infos_ptr = scan_infos;
 	found_board = nullptr;
+	// query in limited IAC - liac - using code from hcitool
+	// utils/tools/hcitool.c
+	const int liac_value = 0x9e8b00; // LIAC (https://www.bluetooth.org/en-us/specification/assigned-numbers/baseband)
```
Comment (delroth) [inline Source/Core/Core/HW/WiimoteReal/IONix.cpp]: Nit: I would just inline that as const uint8_t lap[3] = { 0x00, 0x8b, 0x9e };

I think the comment about this coming from hcitool can be removed given that this is basically a documented constant. However I'd like a comment that explains why this code is using limited IAC (basically a short version of your PR description).

File: Source/Core/Core/HW/WiimoteReal/IONix.cpp [hunk 3]
```diff
@@ -113,14 +118,16 @@ void WiimoteScanner::FindWiimotes(std::vector<Wiimote*> & found_wiimotes, Wiimot
...
-			//	if (wm[j] && bacmp(&scan_infos[i].bdaddr,&wm[j]->bdaddr) == 0)
-			//		new_wiimote = false;
-			//}
+			for (int j = 0; j < MAX_BBMOTES && new_wiimote; ++j)
+			{
+				// compare this address with the stored addresses in our global array
+				// static_cast is OK here, since we're only ever going to have this subclass in g_wiimotes
+				// on Linux (and likewise, only WiimoteWindows on Windows, etc)
+				auto connected_wiimote = static_cast<WiimoteLinux*>(g_wiimotes[j]);
+				if (connected_wiimote && bacmp(&scan_infos[i].bdaddr,&connected_wiimote->Address()) == 0)
```
Comment (delroth) [inline Source/Core/Core/HW/WiimoteReal/IONix.cpp]: space after comma

File: Source/Core/Core/HW/WiimoteReal/IONix.cpp [hunk 4]
```diff
@@ -280,4 +292,5 @@ int WiimoteLinux::IOWrite(u8 const* buf, size_t len)
 	return write(m_int_sock, buf, (int)len);
 }
 
-}; // WiimoteReal
+};
```
Comment (delroth) [inline Source/Core/Core/HW/WiimoteReal/IONix.cpp]: Unnecessary diff.

File: Source/Core/Core/HW/WiimoteReal/IONix.cpp (hunk 2 above)
Comment (delroth) [inline Source/Core/Core/HW/WiimoteReal/IONix.cpp]: (Actually I'd also s/uint8_t/u8/ for consistency with Dolphin's usage of sized integer types -- ours predate C++11 and are still present pretty much all over the codebase.)

**Comment by atahanacar** [comment]
Building caused errors, adding `#include "Common/Logging/Log.h"` back fixed it.

**Comment by Parlane** [comment]
Thank you for your original work, much appreciated. This got submitted as part of #3865
